sdc/datatypes/hpat_pandas_dataframe_functions.py

```diff
@@ -31,64 +31,38 @@
 import pandas
+import numpy
+
+import sdc
 
 from numba import types
 from numba.extending import (overload, overload_method, overload_attribute)
+from sdc.hiframes.pd_dataframe_ext import DataFrameType
 from numba.errors import TypingError
-
-from sdc.datatypes.hpat_pandas_dataframe_types import DataFrameType
-
-
-@overload_method(DataFrameType, 'count')
-def sdc_pandas_dataframe_count(self, axis=0, level=None, numeric_only=False):
-    """
-    Pandas DataFrame method :meth:`pandas.DataFrame.count` implementation.
-
-    .. only:: developer
-
-        Test: python -m sdc.runtests sdc.tests.test_dataframe.TestDataFrame.test_count
-
-    Parameters
-    -----------
-    self: :class:`pandas.DataFrame`
-        input arg
-    axis:
-        *unsupported*
-    level:
-        *unsupported*
-    numeric_only:
-        *unsupported*
-
-    Returns
-    -------
-    :obj:`pandas.Series` or `pandas.DataFrame`
-            returns: For each column/row the number of non-NA/null entries. If level is specified returns a DataFrame.
-    """
-
-    _func_name = 'Method pandas.dataframe.count().'
-
-    if not isinstance(self, DataFrameType):
-        raise TypingError('{} The object must be a pandas.dataframe. Given: {}'.format(_func_name, self))
-
-    if not (isinstance(axis, types.Omitted) or axis == 0):
-        raise TypingError("{} 'axis' unsupported. Given: {}".format(_func_name, axis))
-
-    if not (isinstance(level, types.Omitted) or level is None):
-        raise TypingError("{} 'level' unsupported. Given: {}".format(_func_name, axis))
-
-    if not (isinstance(numeric_only, types.Omitted) or numeric_only is False):
-        raise TypingError("{} 'numeric_only' unsupported. Given: {}".format(_func_name, axis))
-
-    def sdc_pandas_dataframe_count_impl(self, axis=0, level=None, numeric_only=False):
-        result_data = []
-        result_index = []
-
-        for dataframe_item in self._data:
-            item_count = dataframe_item.count()
-            item_name = dataframe_item._name
-            result_data.append(item_count)
-            result_index.append(item_name)
-
-        return pandas.Series(data=result_data, index=result_index)
-
-    return sdc_pandas_dataframe_count_impl
+from sdc.datatypes.hpat_pandas_series_functions import TypeChecker
+
+def sdc_pandas_dataframe_reduce_columns(df, name, params):
+    saved_columns = df.columns
+    n_cols = len(saved_columns)
+    data_args = tuple('data{}'.format(i) for i in range(n_cols))
+    all_params = ['df']
+    for key, value in params:
+        all_params.append('{}={}'.format(key, value))
+    func_definition = 'def _reduce_impl({}):'.format(', '.join(all_params))
+    func_lines = [func_definition]
+    for i, d in enumerate(data_args):
+        line = '  {} = hpat.hiframes.api.init_series(hpat.hiframes.pd_dataframe_ext.get_dataframe_data(df, {}))'
+        func_lines.append(line.format(d + '_S', i))
+        func_lines.append('  {} = {}.{}()'.format(d + '_O', d + '_S', name))
```

Comment:
Why do we call the method of series without any arguments? 
You have to actually forward the arguments passed to DF call into the Series method. And also take care about how they actually map to each other (e.g. for DF.mean and Series.mean arguments are the same, but will it be so for all functions?)

Reply:
Though arguments of df.mean and series.mean are the same, it can't be forwarded straight forward.
1. `axis` for series supports only `index (0)`
2. `numeric_only` is not support by series at all, and shouldn't be forwarded
3.  `level` at the moment we are not storing series in data frame. So there are no hierarchical index and level parameter doesn't make any sense for underlying series, but still there is sense for dataframe. on the other hand at the moment hierarchical index is not support for dataframe either.

https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.mean.html 
https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.mean.html